install/kubernetes/Makefile

```diff
@@ -34,80 +29,37 @@ HELM_DOCS := $(DOCKER_RUN) $(HELM_DOCS_IMAGE)
-		$(SED) -i '/# operator-aws-digest.*/{!b;n;s/awsDigest.*/awsDigest: "'$(OPERATOR_AWS_DIGEST)'"/}' $(CILIUM_VALUES);                 \
-		$(SED) -i '/# operator-azure-digest.*/{!b;n;s/azureDigest.*/azureDigest: "'$(OPERATOR_AZURE_DIGEST)'"/}' $(CILIUM_VALUES);         \
-		$(SED) -i '/# operator-alibabacloud-digest.*/{!b;n;s/alibabacloudDigest.*/alibabacloudDigest: "'$(OPERATOR_ALIBABACLOUD_DIGEST)'"/}' $(CILIUM_VALUES);         \
-		$(SED) -i '/# operator-generic-digest.*/{!b;n;s/genericDigest.*/genericDigest: "'$(OPERATOR_GENERIC_DIGEST)'"/}' $(CILIUM_VALUES); \
-		$(SED) -i '/# clustermesh-apiserver-digest.*/{!b;n;s/digest.*/digest: "'$(CLUSTERMESH_APISERVER_DIGEST)'"/}' $(CILIUM_VALUES)
-
-CRDS = $(foreach path,$(patsubst %.yaml,%,$(shell find $(ROOT_DIR)/examples/crds/*/ -type f)),$(shell basename $(path)))
-lint:
+all: update-versions docs lint
+
+update-chart: $(ROOT_DIR)/VERSION # Update chart versions to point to the current version.
+	$(ECHO_GEN)cilium/Chart.yaml
+	$(QUIET)grep -lR -e 'version:' -e 'appVersion:' $(CILIUM_CHARTS)/ \
+		| xargs -L 1 $(SED) -i -e 's/'$(CILIUM_CHART_REGEX)'/\1 '$(VERSION)'/g'
+	$(QUIET)$(SED) -i 's;icon:.*;icon: $(LOGO_BASE_URL)/cilium@$(CILIUM_BRANCH)/$(LOGO_PATH);' $(CHART_FILE)
+
+cilium/values.yaml: $(CILIUM_CHARTS)/values.yaml.tmpl $(ROOT_DIR)/VERSION $(ROOT_DIR)/$(RELATIVE_DIR)/Makefile.digests $(ROOT_DIR)/$(RELATIVE_DIR)/$(MAKEFILE_VALUES)
+	$(ECHO_GEN)$@
+	echo -e "# File generated by $(RELATIVE_DIR)/Makefile; DO NOT EDIT." > $@
+	echo -e "# This file is based on $(RELATIVE_DIR)/cilium/values.yaml.tmpl.\n" >> $@
+	envsubst < $< >> $@
+
+update-versions: update-chart cilium/values.yaml # Update the Helm values file to point to the current version.
+
+ALL_HELM_FILES := $(shell find cilium/ -type f)
+CRD_FILES := $(shell find $(ROOT_DIR)/examples/crds/*/ -type f)
+CRDS := $(foreach path,$(patsubst %.yaml,%,$(CRD_FILES)),$(shell basename $(path)))
+lint: cilium/values.yaml $(CRD_FILES) $(ALL_HELM_FILES)
+	$(ECHO_CHECK)
 	$(QUIET)for crd in $(CRDS); do \
 		grep -q $$crd $(CHART_FILE) \
 		|| (echo -e "$$crd not found in $(CHART_FILE).\nPlease update the chart to include $$crd."; exit 1); \
 	done
-	$(QUIET)helm lint --with-subcharts --values ./cilium/values.yaml ./cilium
+	$(QUIET)helm lint --with-subcharts --values $< ./cilium
 
 docs:
 	$(QUIET)$(HELM_DOCS)
 
-check-docker-images:
-	$(QUIET)\
-         HUBBLE_UI_VERSION=$(HUBBLE_UI_VERSION) \
-         HUBBLE_UI_BACKEND_VERSION=$(HUBBLE_UI_BACKEND_VERSION) \
-         MANAGED_ETCD_VERSION=$(MANAGED_ETCD_VERSION) \
-         ETCD_VERSION=$(ETCD_VERSION) \
-         NODEINIT_VERSION=$(NODEINIT_VERSION) \
-         CERTGEN_VERSION=$(CERTGEN_VERSION) \
-         ../../contrib/release/check-docker-images.sh "v$(VERSION)"
-
-.PHONY: all check-docker-images clean docs lint update-versions
+check-docker-images: # Check whether docker images are available for the current version.
```

Comment:
running this target in this PR somehow failed for existing images

```
docker.io/cilium/cilium-etcd-operator:v2.0.7@sha256:04b8327f7f992693c2cb483b999041ed8f92efc8e14f2a5f3ab95574a65ea2dc does not exist!
docker.io/cilium/startup-script:d69851597ea019af980891a4628fb36b7880ec26 does not exist!
...
```

Reply:
> Yeah, those images aren't on docker.io

That was actually a lie, it appears that the same digests are just not present on DockerHub. Maybe the more accurate statement would be, "we don't configure the charts to pull those images from DockerHub".

Reply:
I added one more commit on top that should make this work on the master branch now.